troposphere/static/js/components/modals/instance/InstanceLaunchWizardModal.jsx

```diff
@@ -94,9 +99,10 @@ export default React.createClass({
         // Check if the user has any projects, if not then set view to "PROJECT_VIEW"
         // to create a new one
         let projectList = stores.ProjectStore.getAll();
-        if (projectList) {
-            if (view != "IMAGE_VIEW" && projectList.length === 0) {
-                this.viewProject();
+        let sshKey = stores.SSHKeyStore.getAll();
+        if (sshKey) {
+            if (view != "IMAGE_VIEW" && sshKey.length === 0) {
```

Comment:
These two functions serve different purposes. The first one in line 61 is checking whether the user has a public key uploaded or not. This one is trying to have a list of SSH keys back and show that to the user while they are trying to launch instances. I might can have it into one function. Thanks for pointing it out